dags/inference/configs/maxdiffusion_inference_gce_config.py

```diff
@@ -60,25 +72,49 @@ def get_maxdiffusion_inference_nightly_config(
       "cd maxdiffusion",
       "pip3 install jax[tpu] -f https://storage.googleapis.com/jax-releases/libtpu_releases.html",
       "pip3 install -r requirements.txt",
-      "pip3 install ."
+      "pip3 install .",
+      # dependency for controlnet
+      "apt-get install ffmpeg libsm6 libxext6  -y" "cd ..",
   )
 
-  additional_metadata_dict = {
-      "per_device_batch_size": f"{model_configs['per_device_batch_size']}",
-  }
-
-  run_model_cmds = (
-      # Start virtual environment
-      "source .env/bin/activate",
-      ### Benchmark
-      "cd maxdiffusion",
-      # Configure flags
-      """ python -m src.maxdiffusion.generate_sdxl src/maxdiffusion/configs/base_xl.yml run_name="my_run" """,
-      # Give server time to start
-      f"sleep {model_configs['sleep_time']}",
-      f"gsutil cp metrics.json {metric_config.SshEnvVars.GCS_OUTPUT.value}",
-  )
+  if model_name == "SDXL-Base-1.0":
+    run_model_cmds = (
+        # Start virtual environment
+        "source .env/bin/activate",
+        ### Benchmark
+        "cd maxdiffusion",
+        # Configure flags
+        "cd .."
+        f""" python -m src.maxdiffusion.generate_sdxl src/maxdiffusion/configs/base_xl.yml run_name="my_run" per_device_batch_size={per_device_bat_size} attention="{attention}" """,
+        "cd ..",
+        f"gsutil cp metrics.json {metric_config.SshEnvVars.GCS_OUTPUT.value}",
+    )
+  if model_name == "SDXL-Lightning":
+    run_model_cmds = (
+        # Start virtual environment
+        "source .env/bin/activate",
+        ### Benchmark
+        "cd maxdiffusion",
+        # Configure flags
+        "cd .."
+        f""" python -m src.maxdiffusion.generate_sdxl src/maxdiffusion/configs/base_xl.yml run_name="my_run" lightning_repo="ByteDance/SDXL-Lightning" lightning_ckpt="sdxl_lightning_4step_unet.safetensors" per_device_batch_size={per_device_bat_size} attention="{attention}" """,
+        "cd ..",
+        f"gsutil cp metrics.json {metric_config.SshEnvVars.GCS_OUTPUT.value}",
+    )
+  if model_name == "SDXL-ControlNet":
+    run_model_cmds = (
+        # Start virtual environment
+        "source .env/bin/activate",
+        ### Benchmark
+        "cd maxdiffusion",
+        # Configure flags
+        "cd .."
+        f""" python src/maxdiffusion/controlnet/generate_controlnet_sdxl_replicated.py per_device_batch_size={per_device_bat_size} attention="{attention}" """,
+        "cd ..",
+        f"gsutil cp metrics.json {metric_config.SshEnvVars.GCS_OUTPUT.value}",
+    )
 
+  _modify_save_metrics("metrics.json", model_configs)
```

Comment:
I'm not sure if `_modify_save_metrics` will actually affect the `metrics.json` file. We're running the `_modify_save_metrics` function when we define the DAG, but to actually need the function to run in the TPU VM at run time.

We can either have bash commands to do something like `_modify_save_metrics`, which may be more difficult. OR, we could have a python script in the maxdiffusion repo (or other repo) to do this logic. OR, we could do this logic directly in the `generate_sdxl.py` file (with a flag to add the `dimensions` key.